spec/lucky/text_helpers/svg_inliner_spec.cr

```diff
@@ -0,0 +1,37 @@
+require "./text_helpers_spec"
+
+@[Lucky::SvgInliner::Path("spec/fixtures")]
+module Lucky::SvgInliner
+end
+
+class TextHelperTestPage
+  def with_inlined_svg
+    inline_svg("lucky_logo.svg")
+  end
+
+  def with_inlined_and_styled_svg
+    inline_svg("lucky_logo.svg", false)
+  end
+end
+
+describe Lucky::SvgInliner do
+  describe ".inline_svg" do
+    it "inlines an svg in a page" do
+      inlined_svg = view.tap(&.with_inlined_svg).render
+      inlined_svg.should start_with %(<svg data-inline-svg="lucky_logo.svg")
+      inlined_svg.should_not contain %(<?xml version="1.0" encoding="UTF-8"?>)
+      inlined_svg.should_not contain %(<!-- lucky logo -->)
+      inlined_svg.should_not contain "\n"
+      inlined_svg.should_not contain %(fill="none" stroke="#2a2a2a" class="logo")
+    end
+
+    it "inlines an svg in a page with its original styling attributes" do
+      inlined_svg = view.tap(&.with_inlined_and_styled_svg).render
+      inlined_svg.should start_with %(<svg data-inline-svg-styled="lucky_logo.svg")
+      inlined_svg.should contain %(fill="none" stroke="#2a2a2a" class="logo")
+      inlined_svg.should_not contain %(<?xml version="1.0" encoding="UTF-8"?>)
+      inlined_svg.should_not contain %(<!-- lucky logo -->)
+      inlined_svg.should_not contain "\n"
```

Comment:
I like that this strips out un-needed stuff from the XML. Should these `should_not` cases be broken out into an `it "strips out comments"` and `it "strips out the xml header"` set of tests? As it is, if something comes up and someone needs to modify the filter, they don't have a test which directly correlates to that feature.

Reply:
Good point. I'll add separate specs for those cases separately.